### ISSUE TITLE
[Enhancement] Support for default permissions

### ISSUE THREAD

**Issue by Andarius** [wix/Detox#2978]
**Is your feature request related to a problem? Please describe.**

I just added a new package (https://github.com/zo0r/react-native-push-notification) an it became tedious to specify everywhere `launchApp` is specified `{ permissions: { notifications: 'NO'} }` so the pop-up won't show when the app is starting.

**Describe the solution you'd like**
Add support for default permissions that could be overridden when specifying `permissions`.
This could be either in `detox.init` or in the configuration file.

**Describe alternatives you've considered**
Specifying `{ permissions: { notifications: 'NO'} }` everywhere `launchApp` is specified.

**Comment by noomorph** [comment]
Thanks for the feature request. I think we can take it quite soon into the loop.

**Comment by d4vidi** [comment]
@Andarius have you considered creating a wrapper over `detox.launchApp()` that always injects that?

**Comment by d4vidi** [comment]
We are pretty preoccupied with lot's of other things. Let us know if it solves it for  you.

**Comment by Andarius** [comment]
> We are pretty preoccupied with lot's of other things. Let us know if it solves it for you.

No worries, for now I'll stick to `{ permissions: { notifications: 'NO'} }`

**Comment by noomorph** [comment]
@d4vidi  I'm taking it for the bugathon

**Comment by asafkorem** [comment]
@noomorph can you please elaborate on how you're thinking to implement this? 
Why implementing a function that wraps `launchApp` with the additional configurations in the user's end isn't enough? that makes sense to me.

@Andarius, are you still interested in this feature?

**Comment by Andarius** [comment]
@asafkorem yes, I'm still interested :)

**Comment by d4vidi** [comment]
@noomorph so IIUC, you're basically planning on [launch-args-ing](https://github.com/wix/Detox/blob/master/docs/APIRef.LaunchArgs.md#arguments-setup) it 😄

If that means you're planning on adding a runtime editing API on top of the static config, I must say `device.permission` would sound weird (`application.permissions`...). But then again, device.launchArgs sounds almost as weird.

**Comment by noomorph** [comment]
@d4vidi, no, I just plan to add that option to the app config itself. Nothing more than than. No runtime so far yet.
